jbi/actions/steps.py

```diff
@@ -248,10 +249,36 @@ def sync_whiteboard_labels(context: ActionContext, **parameters):
     """
     Set whiteboard tags as labels on the Jira issue
     """
+
+    def whiteboard_as_list(whiteboard: Optional[str]) -> list[str]:
+        splitted = whiteboard.replace("[", "").split("]") if whiteboard else []
+        stripped = [x.strip() for x in splitted if x not in ["", " "]]
+        # Jira labels can't contain a " ", convert to "."
+        nospace = [wb.replace(" ", ".") for wb in stripped]
+        with_brackets = [f"[{wb}]" for wb in nospace]
+        return ["bugzilla"] + nospace + with_brackets
+
+    # On update of whiteboard field, add/remove corresponding labels
+    if context.event.changes:
+        changes_by_field = {change.field: change for change in context.event.changes}
+        if change := changes_by_field.get("whiteboard"):
+            before = whiteboard_as_list(change.removed)
+            after = whiteboard_as_list(change.added)
+            # We don't bother detecting if label was already there.
+            updates = [{"add": label} for label in after]
+            if removed := set(before).difference(set(after)):
+                updates.extend([{"remove": label} for label in sorted(removed)])
```

Comment:
This function is already doing a lot, I wonder if we should extract this into a separate function as well:

```python
def build_issue_labels(added, removed=None):
    before = _whiteboard_as_list(removed)
    after = _whiteboard_as_list(added)
    # We don't bother detecting if label was already there.
    updates = [{"add": label} for label in after]
    if removed := set(before).difference(set(after)):
        updates.extend([{"remove": label} for label in sorted(removed)])
    return updates
```

This is also reusable in the else block for handling creating an issue, I think

Reply:
I refactored, thanks 💯